### PR TITLE
Style: include hover state on contribution buttons

### LINKED ISSUE #73: Amount component: fix button selected and hover state colors
The amount component, used in the join form and the contribution page, is missing hover state in the buttons (monthly/annual and amounts) and has a selected state is different. These buttons follow the "highlighted button outline" style ([link to design system page](https://design.penpot.app/#/view/1555ffb0-ea21-11eb-9b24-4c4b7a420f0b?page-id=1555ffb1-ea21-11eb-9b24-4c4b7a420f0b&index=5&section=interactions)) -- adding a screenshot of the relevant part for quick reference (normal/hover/disabled states):
image

A screenshot of the component in the [contribution page](https://design.penpot.app/#/view/bf28f200-e971-11eb-9b24-4c4b7a420f0b?page-id=36a18270-e972-11eb-b4d3-978aa0906fd6&index=0)
image

### PR DESCRIPTION
**Pull request by basilesimon**
Close #73 

Aims to include a hover state for contribution buttons. 

https://user-images.githubusercontent.com/5701152/153021511-908a0352-e34b-4b6a-8406-aceda891836e.mov

### CODE REVIEW

File: src/components/contribution/ContributionAmount.vue [hunk 1]
```diff
@@ -93,7 +90,11 @@
           justify-center
           text-primary
           font-semibold
+          hover:bg-link-10
         "
+        :class="{
+          'bg-link text-white hover:bg-link-110': amount === modelValue,
```
Comment (wpf500) [inline src/components/contribution/ContributionAmount.vue:96]: I think `hover:bg-link-110` probably just wants to be `hover:bg-link` to match the change in `ContributionPeriod`?

Alternatively so you don't have to override the hover state you could conditionally set `hover:bg-link-10`, e.g.
```
:class="amount === modelValue ? 'bg-link text-white' : 'hover:bg-link-10'
```

File: src/components/contribution/ContributionPeriod.vue [hunk 2]
```diff
@@ -23,9 +23,10 @@
           font-semibold
           h-10
           cursor-pointer
+          hover:bg-link-10
         "
         :class="{
-          'bg-link text-white': period.name === selectedPeriod,
+          'bg-link text-white hover:bg-link': period.name === selectedPeriod,
```
Comment (wpf500) [inline src/components/contribution/ContributionPeriod.vue:29]: Could potentially use the same conditional logic as above

**Review by wpf500 — CHANGES_REQUESTED** [review]

**Comment by basilesimon** [comment]
@wpf500 For some reason I'm not able to run the repo at the minute (the `value.roles` TypeError), but your comments make sense – are you able to apply them?

**Comment by wpf500** [comment]
@basilesimon yup I'll add them in and merge :+1:. The `roles` error is just because I've now merged all the changes I mentioned before so this branch is out of date with the API on dev.

**Comment by wpf500** [comment]
Closing in favour of https://github.com/beabee-communityrm/beabee-frontend/pull/98 as I can't edit your forked branch

**Pull request closed without merging**